testsuite/tests/contrib/lkt_semantic/decl_block_correctness/test.out

```diff
@@ -0,0 +1,223 @@
+
+Id   <RefId "T" test.lkt:12:35-12:36>
+     references <GenericFormalTypeDecl "T" test.lkt:12:12-12:13>
+
+Id   <RefId "i" test.lkt:12:39-12:40>
+     references <FunArgDecl "i" test.lkt:12:28-12:32>
+
+Expr <RefId "i" test.lkt:12:39-12:40>
+     has type <GenericFormalTypeDecl "T" test.lkt:12:12-12:13>
+
+test.lkt:18:4: error: value declaration forbidden in enum declaration
+17 |    @invalid val i: Int = 1
+   |    ^^^^^^^^^^^^^^^^^^^^^^^
+
+test.lkt:19:4: error: generic class declaration forbidden in enum declaration
+18 |    @invalid generic[T] class MyGenClass {}
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Id   <RefId "Int" test.lkt:18:20-18:23>
+     references <StructDecl prelude: "Int">
+
+Expr <NumLit test.lkt:18:26-18:27>
+     has type <StructDecl prelude: "Int">
+
+test.lkt:23:4: error: value declaration forbidden in struct declaration
+22 |    @invalid val i: Int = 1
+   |    ^^^^^^^^^^^^^^^^^^^^^^^
+
+test.lkt:24:4: error: struct declaration forbidden in struct declaration
+23 |    @invalid struct InnerStruct {}
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+test.lkt:25:4: error: generic value declaration forbidden in struct declaration
+24 |    @invalid generic[T] fun MyGenFun(i: T): T = i
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Id   <RefId "Int" test.lkt:23:20-23:23>
+     references <StructDecl prelude: "Int">
+
+Expr <NumLit test.lkt:23:26-23:27>
+     has type <StructDecl prelude: "Int">
+
+Id   <RefId "T" test.lkt:25:40-25:41>
+     references <GenericFormalTypeDecl "T" test.lkt:25:21-25:22>
+
+Id   <RefId "T" test.lkt:25:44-25:45>
+     references <GenericFormalTypeDecl "T" test.lkt:25:21-25:22>
+
+Id   <RefId "i" test.lkt:25:48-25:49>
+     references <FunArgDecl "i" test.lkt:25:37-25:41>
+
+Expr <RefId "i" test.lkt:25:48-25:49>
+     has type <GenericFormalTypeDecl "T" test.lkt:25:21-25:22>
+
+Id   <RefId "Int" test.lkt:27:17-27:20>
+     references <StructDecl prelude: "Int">
+
+Id   <RefId "Int" test.lkt:27:23-27:26>
+     references <StructDecl prelude: "Int">
+
+Id   <RefId "i" test.lkt:27:29-27:30>
+     references <FunArgDecl "i" test.lkt:27:14-27:20>
+
+Expr <RefId "i" test.lkt:27:29-27:30>
+     has type <StructDecl prelude: "Int">
+
+Id   <RefId "Int" test.lkt:28:11-28:14>
+     references <StructDecl prelude: "Int">
+
+Expr <NumLit test.lkt:28:17-28:18>
+     has type <StructDecl prelude: "Int">
+
+test.lkt:32:4: error: value declaration forbidden in class declaration
+31 |    @invalid val i: Int = 1
+   |    ^^^^^^^^^^^^^^^^^^^^^^^
+
+test.lkt:33:4: error: class declaration forbidden in class declaration
+32 |    @invalid class InnerClass {}
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+test.lkt:34:4: error: generic trait declaration forbidden in class declaration
+33 |    @invalid generic[T] trait MyTrait {}
+   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+test.lkt:35:4: error: generic value declaration forbidden in class declaration
```

Comment:
This is not a value declaration, but a field declaration (same below)

Reply:
Good catch!